### PR TITLE
strerr: Return error msg instead of NULL on invalid errnum

### PR DESCRIPTION
**Pull request by agkaminski**
DONE: RTOS-418

## Description

Fixes https://github.com/phoenix-rtos/phoenix-rtos-project/issues/668
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [x] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
CI fail is caused by stale psu in the docker image

**Review by gerard5 — APPROVED** [review]
The solution works as it should.  MISRA remarks, not a merge blocker.

File: string/strerror.c [hunk 1]
```diff
@@ -42,27 +44,36 @@ static const errinfo_t gaitab[] = {
...
 	e = bsearch((void *)(long)errnum, tab, tabsz / sizeof(*tab), sizeof(*tab), err_cmp);
 
-	return e != NULL ? names + e->offset : NULL;
+	if (e == NULL) {
+		(void)sprintf(err_unknownMsg, "Unknown error %d", errnum);
+		errno = EINVAL;
+		return err_unknownMsg;
+	}
+	else {
+		return names + e->offset;
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply string/strerror.c:75]: This rule is advisory

File: string/strerror.c [hunk 2]
```diff
@@ -42,27 +44,36 @@ static const errinfo_t gaitab[] = {
...
+static char err_unknownMsg[32]; /* FIXME: Make thread safe? IMHO overkill */
 
 
 static int err_cmp(const void *key, const void *elem)
 {
 	const errinfo_t *p = elem;
 	long err = (long)key;
 
-	return err - p->errno;
+	return err - p->errnum;
```
[the remark this answers is not in the document]
Comment (agkaminski) [reply string/strerror.c]: Done. Let's ignore more forbidden `const void *` to `long` cast :face_in_clouds:

[the remark this answers is not in the document]
Comment (gerard5) [reply string/strerror.c]: Yep, that's true. But it's tricky to resolve it for that use case and it certainly would complicate the readability.

File: string/strerror.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (gerard5) [reply string/strerror.c:75]: Ack.

**Pull request merged**